File: Seafood Guide/Data Model/FishCat.m
//
//  FishCat.m
//  Seafood Guide
//
//  Created by Jon Brown on 9/2/14.
//  Copyright (c) 2014 Jon Brown Designs. All rights reserved.
//

#import "FishCat.h"


@implementation FishCat

@synthesize window = _window;
@synthesize caption, image;
@synthesize fetchedResultsController = __fetchedResultsController;
@synthesize managedObjectContext = __managedObjectContext;

-(id)initWithCaption:(NSString*)theCaption andImage:(UIImage*)theImage
{
    self = [super init];
    
    if(self)
    {
        
    }
    self.caption = theCaption;
    self.image = theImage;
    
    return self;
}

+(NSArray*)getSampleData
{
    
    FishCat* fish1 = [[FishCat alloc] initWithCaption:@"Mild Fish" andImage:[UIImage imageNamed:@"fish1"]];
    
    FishCat* fish2 = [[FishCat alloc] initWithCaption:@"Thicker & Flavorful" andImage:[UIImage imageNamed:@"fish2"]];
    
    FishCat* fish3 = [[FishCat alloc] initWithCaption:@"Steak Like Fish" andImage:[UIImage imageNamed:@"fish3"]];
    
    FishCat* fish4 = [[FishCat alloc] initWithCaption:@"Small Fish" andImage:[UIImage imageNamed:@"fish4"]];
    
    FishCat* fish5 = [[FishCat alloc] initWithCaption:@"Shell Fish" andImage:[UIImage imageNamed:@"fish5"]];
    
    FishCat* fish6 = [[FishCat alloc] initWithCaption:@"Other" andImage:[UIImage imageNamed:@"fish6"]];
    
    FishCat* fish7 = [[FishCat alloc] initWithCaption:@"All Fish" andImage:[UIImage imageNamed:@"allfish.png"]];
    
    FishCat* fish8 = [[FishCat alloc] initWithCaption:@"Dirty Dozen" andImage:[UIImage imageNamed:@"donteat"]];
    
    
    return [NSArray arrayWithObjects:fish1, fish2, fish3, fish4, fish5, fish6, fish7, fish8, nil];
}


@end

File: Seafood Guide/Data Model/FishCat.swift
//  Converted to Swift 5.1 by Swiftify v5.1.26565 - https://objectivec2swift.com/
//
//  FishCat.swift
//  Seafood Guide
//
//  Created by Jon Brown on 9/2/14.
//  Copyright (c) 2014 Jon Brown Designs. All rights reserved.
//

import CoreData
import Foundation
import UIKit

@objcMembers class FishCat: NSObject {
    var window: UIWindow?
    var caption = ""
    var image: UIImage?
    var fetchedResultsController: NSFetchedResultsController<NSFetchRequestResult>?
    var managedObjectContext: NSManagedObjectContext?

    init(caption theCaption: String?, andImage theImage: UIImage?) {
        super.init()
        caption = theCaption ?? ""
        image = theImage
    }

    class func getSampleData() -> [AnyHashable]? {

        let fish1 = FishCat(caption: "Mild Fish", andImage: UIImage(named: "fish1"))

        let fish2 = FishCat(caption: "Thicker & Flavorful", andImage: UIImage(named: "fish2"))

        let fish3 = FishCat(caption: "Steak Like Fish", andImage: UIImage(named: "fish3"))

        let fish4 = FishCat(caption: "Small Fish", andImage: UIImage(named: "fish4"))

        let fish5 = FishCat(caption: "Shell Fish", andImage: UIImage(named: "fish5"))

        let fish6 = FishCat(caption: "Other", andImage: UIImage(named: "fish6"))

        let fish7 = FishCat(caption: "All Fish", andImage: UIImage(named: "allfish.png"))

        let fish8 = FishCat(caption: "Dirty Dozen", andImage: UIImage(named: "donteat"))


        return [fish1, fish2, fish3, fish4, fish5, fish6, fish7, fish8]
    }
}
